pkg/graveler/sstable/writer.go

```diff
@@ -138,10 +138,14 @@ func (dw *DiskWriter) Close() (*committed.WriteResult, error) {
 	// Prepare metadata properties for Close to write.  The map was already set in the
 	// sstable.Writer constructor and cannot be changed, but we can replace its values
 	// before writing it out.
-	dw.SetMetadata(MetadataFirstKey, string(dw.first))
-	dw.SetMetadata(MetadataLastKey, string(dw.last))
-	dw.SetMetadata(MetadataNumRecordsKey, fmt.Sprint(dw.count))
-	dw.SetMetadata(MetadataEstimatedSizeKey, fmt.Sprint(dw.w.EstimatedSize()))
+	first := dw.first
+	last := dw.last
+	estimatedSize := dw.w.EstimatedSize()
+	count := dw.count
```

Comment:
Why set these values now? Does the `Store`/`Close` change them?

Reply:
Yes - panic. The `EstimatedSize` calculate and access underlaying structure that we already closed.